lib/cn/kvset_builder.c

```diff
@@ -37,31 +38,35 @@ kvset_builder_create(
     struct kvset_builder *bld;
     merr_t                err;
 
-    bld = malloc(sizeof(*bld));
+    bld = calloc(1, sizeof(*bld));
     if (!bld)
         return merr(ev(ENOMEM));
 
-    memset(bld, 0, sizeof(*bld));
-
     bld->seqno_min = U64_MAX;
 
-    err = kbb_create(&bld->kbb, cn, pc);
+    err = hbb_create(&bld->hbb, cn);
     if (ev(err))
         goto err_exit1;
 
-    err = vbb_create(&bld->vbb, cn, pc, vgroup);
+    err = kbb_create(&bld->kbb, cn, pc);
     if (ev(err))
         goto err_exit2;
 
+    err = vbb_create(&bld->vbb, cn, pc, vgroup);
+    if (ev(err))
+        goto err_exit3;
+
     bld->cn = cn;
     bld->key_stats.seqno_prev = U64_MAX;
     bld->key_stats.seqno_prev_ptomb = U64_MAX;
 
     *bld_out = bld;
     return 0;
 
-err_exit2:
+err_exit3:
     kbb_destroy(bld->kbb);
+err_exit2:
+    hbb_destroy(bld->hbb);
```

Comment:
Combine the err_exit labels into one since the destroy methods are safe wrt to NULL ptrs.

Reply:
I hate "exit", sounds like you're going to call exit()...